cmd/tempo-cli/cmd-list-block.go

```diff
@@ -85,7 +85,7 @@ func dumpBlock(r tempodb_backend.Reader, c tempodb_backend.Compactor, tenantID s
 	if scan {
 		fmt.Println("Scanning block contents.  Press CRTL+C to quit ...")
 
-		block, err := encoding.NewBackendBlock(&unifiedMeta.BlockMeta, r)
+		block, err := v2.NewBackendBlock(&unifiedMeta.BlockMeta, r)
```

Comment:
based on the PR comment I think we're on the same page here. Ultimately this will call `encoding.NewBackendBlock()` and that code will use the meta to determine which object to return?

Reply:
I think there are several options here, basically cosmetic.  

1. We can do `encoding.NewBackendBlock` which returns a new `BackendBlock` interface, but that creates the do-it-all interface that I was trying to avoid.  
2. `encoding.SearchBlock` returns `Searcher`, etc. The smaller interfaces here seem better for readability and testability, but maybe cluttered if there are too many.
3. Or maybe this should all be in `tempodb`.  I don't think `encoding` was intended to be the entry-point to the backend, but it's trending that way.

Was leaving these decisions for the next PR but happy to tackle some now if we can.

Reply:
Nah, feel free to move forward with this PR as is. I think I'd favor #2, but that's just a gut call w/o digging into the details.